DigitalLearningSolutions.Web/Views/TrackingSystem/Delegates/DelegateProgress/EditSupervisor.cshtml

```diff
@@ -0,0 +1,66 @@
+﻿@inject IConfiguration Configuration
+@using DigitalLearningSolutions.Web.Models.Enums
+@using DigitalLearningSolutions.Web.ViewModels.TrackingSystem.Delegates.DelegateProgress
+@using Microsoft.Extensions.Configuration
+@model EditSupervisorViewModel
+
+@{
+  var errorHasOccured = !ViewData.ModelState.IsValid;
+  ViewData["Title"] = errorHasOccured ? "Error: Edit supervisor" : "Edit supervisor";
+  var routeParamsForBackLink = new Dictionary<string, string?>();
+
+  if (Model.AccessedVia.Equals(DelegateProgressAccessRoute.CourseDelegates)) {
+    routeParamsForBackLink.Add("accessedVia", Model.AccessedVia.ToString());
+    routeParamsForBackLink.Add("progressId", Model.ProgressId.ToString());
+  } else {
+    routeParamsForBackLink.Add("delegateId", Model.DelegateId.ToString());
+  }

```

Comment:
Feels like since this only depends on the model it could be set inside it, but I can also see that we already do it like this elsewhere so we might have already decided this is better. Any thoughts?

Reply:
I think it's probably better to leave it as is for now? It's the same pattern as pretty much every other page with a cancel/back link that requires a route parameter. Unless we want to start a new code style here, and tell everyone to start putting this on the model instead?

Reply:
Yeah I'm not sure it's worth it then. Happy to leave it like it is.